test/jdk/java/foreign/StdLibTest.java

```diff
@@ -387,7 +395,7 @@ enum PrintfArg {
         INT(int.class, C_INT, "%d", "%d", arena -> 42, 42),
         LONG(long.class, C_LONG_LONG, "%lld", "%d", arena -> 84L, 84L),
         DOUBLE(double.class, C_DOUBLE, "%.4f", "%.4f", arena -> 1.2345d, 1.2345d),
-        STRING(MemorySegment.class, C_POINTER, "%s", "%s", arena -> arena.allocateFrom("str"), "str");
+        STRING(MemorySegment.class, C_POINTER, "%s", "%s", arena -> arena.allocateFrom("str", Charset.forName(System.getProperty("native.encoding"))), "str");
```

Comment:
The native charset should be captured in a static field and `allocateFrom()` should be used both here and for the format string.

Reply:
Yes, was working on verifying and making changes, made changes in https://github.com/ibmruntimes/openj9-openjdk-jdk/compare/3a1849e72a67997533198aa87ef88cff848560ec..e23794e3f1100f1efcaee135d6a1c7304cf0ea9e to address this. I did verify changes with JDK23 (https://github.com/r30shah/openj9-openjdk-jdk23/commit/56a31ebee44e6897d7972dee45cef5c5cfc94b04) with grinder : `/view/Test_grinder/job/Grinder/43866/`